src/main/ruby/core/range.rb

```diff
@@ -187,7 +187,7 @@ def hash
     hash ^= self.end.hash << 9
     hash ^= excl << 24;
     # Are we throwing away too much here for a good hash value distribution?
-    hash & Fixnum::MAX
+    hash & 0x7fffffffffffffff
```

Comment:
I don't like the magic value here, but this whole method has magic values so maybe it doesn't matter.

Reply:
We could have a `LONG_MAX` constant somewhere, but where?

Reply:
In the `Truffle` module maybe? Maybe a comment here is good enough.

Reply:
Added under `Truffle::Platform`.